components/locid_transform/src/expander.rs

```diff
@@ -101,19 +102,68 @@ impl LocaleExpander {
     /// </div>
     pub fn try_new_unstable<P>(provider: &P) -> Result<LocaleExpander, LocaleTransformError>
     where
-        P: DataProvider<LikelySubtagsV1Marker> + ?Sized,
+        P: DataProvider<LikelySubtagsForLanguageV1Marker>
+            + DataProvider<LikelySubtagsForScriptRegionV1Marker>
+            + ?Sized,
     {
-        let likely_subtags: DataPayload<LikelySubtagsV1Marker> =
-            provider.load(Default::default())?.take_payload()?;
+        let likely_subtags_l = provider.load(Default::default())?.take_payload()?;
+        let likely_subtags_sr = provider.load(Default::default())?.take_payload()?;
 
-        Ok(LocaleExpander { likely_subtags })
+        Ok(LocaleExpander {
+            likely_subtags_l,
+            likely_subtags_sr,
+        })
     }
 
-    icu_provider::gen_any_buffer_constructors!(
-        locale: skip,
-        options: skip,
-        error: LocaleTransformError
-    );
+    fn try_new_compat<P>(provider: &P) -> Result<LocaleExpander, LocaleTransformError>
+    where
+        P: DataProvider<LikelySubtagsForLanguageV1Marker>
+            + DataProvider<LikelySubtagsForScriptRegionV1Marker>
+            + DataProvider<LikelySubtagsV1Marker>
+            + ?Sized,
+    {
+        let payload_l = provider
+            .load(Default::default())
+            .and_then(DataResponse::take_payload);
+        let payload_sr = provider
+            .load(Default::default())
+            .and_then(DataResponse::take_payload);
+
+        let (likely_subtags_l, likely_subtags_sr) = if payload_l.is_err() || payload_sr.is_err() {
+            let result: DataPayload<LikelySubtagsV1Marker> =
+                provider.load(Default::default())?.take_payload()?;
+            (
+                payload_l.unwrap_or_else(|_e| {
+                    result.map_project_cloned(|st, _| {
+                        LikelySubtagsForLanguageV1::clone_from_borrowed(st)
+                    })
+                }),
+                payload_sr.unwrap_or_else(|_e| result.map_project(|st, _| st.into())),
+            )
+        } else {
+            (payload_l?, payload_sr?)
```

Comment:
suggestion: if you'd like you can invert the polarity of this `if` and do `if let (Ok(l), Ok(r)) = (payload_l, payload_r)` to avoid double-checking the error bit (though that's almost certainly going ot be optimized)

Reply:
Ok. I did it with `match` so that I can retain ownership of the payloads in the other case.